lib/editor/components/CreateSnapshotModal.js

```diff
@@ -72,7 +102,10 @@ class CreateSnapshotModal extends Component<Props, State> {
 
   render () {
     const {Body, Footer, Header, Title} = Modal
-    const {comment, name, publishNewVersion, showModal} = this.state
+    const {comment, name, publishNewVersion, confirmPublishWithUnapproved, showModal, loading} = this.state
+    const {routes} = this.props
+    const unapprovedRoutes = routes.filter(route => route.hasOwnProperty('status') && route.status !== routeStatusTextToValue('Approved'))
```

Comment:
Same comment as earlier -- we should probably move `routeStatusTextToValue` to return some sort of dictionary.